syntax = "proto3";
package bci;

option go_package = "github.com/tendermint/tendermint/abci/types";


 message RequestSendData {
   uint32  code = 1;
   string log = 2;
   string info = 3;
 }

 message ResponseSendData {
   uint32 code = 1;
   string log = 2;
   string info = 3;
 }


 message RequestJoinRC {
   uint32 code = 1;
   string log = 2;
   string message = 3;
   string rc = 4;
 }

 message ResponseJoinRC {
   uint32 code = 1;
   string log = 2;
   string message = 3;
 }

 message Chain {
   uint32 identifier = 1;
 }

message RequestNextRC {
   repeated Chain target = 1;
}

message ResponseNextRC {
   uint32 code = 1;
   string info = 2;
}

message RequestCallBack {
   repeated Chain source = 1;
}

message ResponseCallBack {
   uint32 code = 1;
   string info = 2;
}

 service Lane {
   rpc SendData(RequestSendData) returns (ResponseSendData);
   rpc JoinRC(RequestJoinRC) returns (ResponseJoinRC);
   rpc NextRC(RequestNextRC) returns (ResponseNextRC);
   rpc CallBack(RequestCallBack) returns (ResponseCallBack);
 }
